### ISSUE TITLE
Enhancement: [promise-reject-error] Support `TypeOrValueSpecifier` format

### ISSUE THREAD

**Issue by christopher-buss** [typescript-eslint/typescript-eslint#12048]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-promise-reject-errors/

### Description

[only-throw-error](https://typescript-eslint.io/rules/only-throw-error/) supports the ability to add a TypeOrValueSpecifier to provide custom Error objects.

I'm requesting feature parity with the other rule, so both of these can be used simultaneously and not error for different things.

### Fail

```TypeScript
-
```

### Pass

```TypeScript
import { Error } from "some-package";

Promsie.reject(new Error("fail"));
```

**Comment by JoshuaKGoldberg** [comment]
👍 that makes sense to me. Normally I'd try to think more deeply on real-world use cases, but given that these two rules are essentially siblings, I agree that it's odd to have one support the format but not the other.

**Comment by christopher-buss** [comment]
My personal use cases are I use a different compiler to TypeScript (not targeting javascript), and as such a lot of the built-ins (like Error object) have to be polyfill'ed through libraries!

**Comment by darshjme-codes** [comment]
**Triage note:**

Enhancement request: Add `TypeOrValueSpecifier` format support to `promise-reject-error` rule's `allowedRejectionTypes` option.

**Current:** Only accepts string type names  
**Proposed:** Allow importing types and referencing them, similar to how `consistent-type-imports` works

**Example:**
```json
{
  "allowedRejectionTypes": [
    {"from": "my-lib", "name": "CustomError"}
  ]
}
```

**Benefits:** More precise type checking, especially for custom error types from libraries

**Labels already applied:** `enhancement: plugin rule option`, `accepting prs` ✓
